homeassistant/components/climate/daikin.py

```diff
@@ -0,0 +1,275 @@
+        """Return the temperature we try to reach."""
+        self._target_temperature = self.settings('stemp', True)
+
+        return self._target_temperature
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        settings = {}
+
+        if kwargs.get(ATTR_OPERATION_MODE) is not None:
+            operation_mode = kwargs.get(ATTR_OPERATION_MODE)
+            current_operation = HA_STATE_TO_DAIKIN.get(operation_mode)
+            if current_operation is not None:
+                settings['mode'] = current_operation
+
+        if kwargs.get(ATTR_TEMPERATURE) is not None:
+            temperature = kwargs.get(ATTR_TEMPERATURE)
+            try:
+                self._target_temperature = int(temperature)
+                settings['stemp'] = str(self._target_temperature)
+            except ValueError:
+                _LOGGER.error("Invalid temperature %s", temperature)
+
+        if settings:
+            self._device.set(settings)
+            self.schedule_update_ha_state()
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        operation_mode = self._device.represent('mode')[1]
+        current_operation = DAIKIN_STATE_TO_HA.get(operation_mode)
+
+        return current_operation
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._operation_list
+
+    def set_operation_mode(self, operation_mode):
+        """Set HVAC mode (auto, auxHeatOnly, cool, heat, off)."""
+        current_operation = HA_STATE_TO_DAIKIN.get(operation_mode)
+
+        if current_operation is not None:
+            self._device.set({"mode": current_operation})
+            self.schedule_update_ha_state()
+        else:
+            _LOGGER.error("Invalid operation mode %s", operation_mode)
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        self._current_humidity = self._device.represent('shum')[1]
+
+        return self._current_humidity
+
+    def set_humidity(self, humidity):
+        """Set new target temperature."""
+        if humidity is not None:
+            try:
+                self._target_humidity = int(humidity)
+                self._device.set({"shum": str(humidity)})
+                self.schedule_update_ha_state()
+            except ValueError:
+                _LOGGER.error("Invalid humidity %s", humidity)
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        self._current_fan_mode = self._device.represent('f_rate')[1]
+
+        return self._current_fan_mode.title()
+
+    def set_fan_mode(self, fan):
+        """Set fan mode."""
+        if fan is not None:
+            self._device.set({"f_rate": fan.lower()})
```

Comment:
What does the device do if I call this function with "foobar" as `fan` parameter? Probably not go up in flames, but I would still sanitize here and check whether `fan` is an allowed string.